From a139a8ee12c2034092485b8b63bc7bbd6bc972ab Mon Sep 17 00:00:00 2001
From: Tingguo Cheng <quic_tingguoc@quicinc.com>
Date: Mon, 28 Oct 2024 10:33:48 +0800
Subject: [PATCH 16/68] arm64: dts: qcom: qcs615: Adds SPMI support

Add the SPMI bus Arbiter node for the PMIC on QCS615 platforms.

Signed-off-by: Tingguo Cheng <quic_tingguoc@quicinc.com>
---
 arch/arm64/boot/dts/qcom/qcs615.dtsi | 29 ++++++++++++++++++++++++++--
 1 file changed, 27 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/qcs615.dtsi b/arch/arm64/boot/dts/qcom/qcs615.dtsi
index c72759c4f28d..ec02015747e4 100644
--- a/arch/arm64/boot/dts/qcom/qcs615.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcs615.dtsi
@@ -553,8 +553,33 @@ sram@c3f0000 {
 			reg = <0x0 0x0c3f0000 0x0 0x400>;
                 };
 
-		apps_smmu: iommu@15000000 {
-			compatible = "qcom,qcs615-smmu-500", "qcom,smmu-500", "arm,mmu-500";
+                spmi_bus: spmi@c440000 {
+                	compatible = "qcom,spmi-pmic-arb";
+                	reg = <0x0 0x0c440000 0x0 0x1100>,
+                	      <0x0 0x0c600000 0x0 0x2000000>,
+                	      <0x0 0x0e600000 0x0 0x100000>,
+                	      <0x0 0x0e700000 0x0 0xa0000>,
+                	      <0x0 0x0c40a000 0x0 0x26000>;
+                	reg-names = "core",
+                	            "chnls",
+                	            "obsrvr",
+                	            "intr",
+                	            "cnfg";
+                	interrupts-extended = <&pdc 1 IRQ_TYPE_LEVEL_HIGH>;
+                	interrupt-names = "periph_irq";
+                	interrupt-controller;
+                	#interrupt-cells = <4>;
+                	#address-cells = <2>;
+                	#size-cells = <0>;
+                	cell-index = <0>;
+                	qcom,channel = <0>;
+                	qcom,ee = <0>;
+               	};
+
+               	apps_smmu: iommu@15000000 {
+               		compatible = "qcom,qcs615-smmu-500",
+               		             "qcom,smmu-500",
+               		             "arm,mmu-500";
 			reg = <0x0 0x15000000 0x0 0x80000>;
 			#iommu-cells = <2>;
 			#global-interrupts = <1>;
-- 
2.34.1

